packages/api/lib/writeRecords/write-granules.js

```diff
@@ -536,38 +501,66 @@ const writeGranules = async ({
   // Process each granule in a separate transaction via Promise.allSettled
   // so that they can succeed/fail independently
   const results = await Promise.allSettled(granules.map(
-    (granule) => _writeGranule({
-      collectionId,
-      granule,
-      processingTimeInfo,
-      error,
-      executionUrl,
-      pdrName,
-      provider,
-      workflowStartTime,
-      workflowStatus,
-      queryFields,
-      collectionCumulusId,
-      providerCumulusId,
-      executionCumulusId,
-      pdrCumulusId,
-      knex,
-      granuleModel,
-    })
+    async (granule) => {
+      // compute granule specific data.
+      const files = await _generateFilesFromGranule({ granule, provider });
+      const timeToArchive = getGranuleTimeToArchive(granule);
+      const timeToPreprocess = getGranuleTimeToPreprocess(granule);
+      const productVolume = getGranuleProductVolume(files);
+      const now = Date.now();
+      const duration = getWorkflowDuration(workflowStartTime, now);
+      const status = getGranuleStatus(workflowStatus, granule);
+      const updatedAt = now;
+
+      const dynamoGranuleRecord = await granuleModel.generateGranuleRecord({
+        granule,
+        executionUrl,
+        collectionId,
+        provider: provider.id,
+        workflowStartTime,
+        files,
+        error,
+        pdrName,
+        workflowStatus,
+        timeToArchive,
+        timeToPreprocess,
+        productVolume,
+        duration,
+        status,
+        processingTimeInfo,
+        queryFields,
+        updatedAt,
+      });
+
+      const postgresGranuleRecord = await translateApiGranuleToPostgresGranule(
+        dynamoGranuleRecord,
+        knex
+      );
+
+      return _writeGranule({
+        postgresGranuleRecord,
+        dynamoGranuleRecord,
+        collectionCumulusId,
```

Comment:
Unnnecessary now, I believe

Reply:
Yeah, I missed that, and that should propagate up also, to not being needed in the sf-event-sqs-to-db-records call to writeGranulesFromMessage.  Which makes sense since it's just the same as writeGranulesFromApi.

Reply:
6e6a9eb5d1